tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,241 @@
+"""Test the BryantEvolutionClient type."""
+
+from datetime import timedelta
+import logging
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.bryant_evolution.climate import SCAN_INTERVAL
+from homeassistant.components.climate import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_ACTION,
+    ATTR_HVAC_MODE,
+    DOMAIN as CLIMATE_DOMAIN,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    HVACAction,
+)
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def trigger_polling(hass: HomeAssistant, freezer: FrozenDateTimeFactory) -> None:
+    """Trigger a polling event."""
+    freezer.tick(SCAN_INTERVAL + timedelta(seconds=1))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+
+async def test_setup_integration_success(
+    hass: HomeAssistant,
+    mock_evolution_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test that an instance can be constructed."""
+    state = hass.states.get("climate.system_1_zone_1")
+    assert state == snapshot
```

Comment:
You can use the `snapshot_platform` helper here, which will snapshot all entities and their states, also capturing attributes and entity options